cdap-master/src/main/java/co/cask/cdap/data/tools/ReplicationStatusTool.java

```diff
@@ -378,13 +442,85 @@ private static void dumpReplicationStateTable() throws Exception {
+    boolean complete;
+    SortedMap<String, String> slaveChecksumMap = getClusterChecksumMap();
+
+    //Check if all files are accounted on both clusters
+    if (masterChecksumMap.size() != slaveChecksumMap.size()) {
+      System.out.println("Number of HDFS files on the Master and Slave Clusters do not match.");
+    }
+
+    complete = !checkDifferences(masterChecksumMap.keySet(), slaveChecksumMap.keySet(), "File");
+
+    for (Map.Entry<String, String> checksumEntry : masterChecksumMap.entrySet()) {
+      String file = checksumEntry.getKey();
+      String masterChecksum = checksumEntry.getValue();
+      String slaveChecksum = slaveChecksumMap.get(file);
+
+      if (slaveChecksum != null && !masterChecksum.equals(slaveChecksum)) {
+        System.out.println("Master Checksum " + masterChecksum + " for File " + file + " does not match with"
+                             + " Slave Checksum " + slaveChecksum);
+        complete = false;
+      }
     }
+
+    if (complete) {
+      // If checksums match for all files.
+      System.out.println("Master and Slave Checksums match. HDFS Replication is complete.");
+    } else {
+      System.out.println("HDFS Replication is NOT Complete.");
+    }
+  }
+
+  private static String normalizedFileName(String fileName) {
+    //while matching the filenames, ignore the IP addresses of the clusters
+    String normalizedName;
+    String ipv4Pattern = "(([01]?\\d\\d?|2[0-4]\\d|25[0-5])\\.){3}([01]?\\d\\d?|2[0-4]\\d|25[0-5])";
+    String ipv6Pattern = "([0-9a-f]{1,4}:){7}([0-9a-f]){1,4}";
+    normalizedName = fileName.replaceFirst(ipv4Pattern, "<hostname>");
+    normalizedName = normalizedName.replaceFirst(ipv6Pattern, "<hostname>");
+    return normalizedName;
   }
 
-  private static String getClusterChecksum() throws IOException {
-    return "checksum";
+  private static SortedMap<String, String> getClusterChecksumMap() throws IOException {
+    FileSystem fileSystem = FileSystem.get(hConf);
+    List<String> fileList = getAllFiles(fileSystem);
+    SortedMap<String, String> checksumMap = new TreeMap<String, String>();
+    for (String file : fileList) {
+      FileChecksum fileChecksum = fileSystem.getFileChecksum(new Path(file));
+      checksumMap.put(normalizedFileName(file), fileChecksum.toString());
+    }
+    LOG.info("Added " + checksumMap.size() + " checksums for snapshot files.");
+    return checksumMap;
+  }
+
+  private static List<String> getAllFiles(FileSystem fs) throws IOException {
+    List<String> fileList = new ArrayList<String>();
+    for (String dirPathName : allChecksumPaths) {
+      Path dirPath = new Path(dirPathName);
+      LOG.info("Getting all files under " + dirPath);
+      getAllDirFiles(dirPath, fs, fileList);
+    }
+    return fileList;
+  }
+
+  private static void getAllDirFiles(Path filePath, FileSystem fs, List<String> fileList) throws IOException {
```

Comment:
nit: addAllDirFiles is a better name, get sounds like it should return something.